hexa/pipelines/models.py

```diff
@@ -131,16 +133,31 @@ def update_if_has_perm(self, principal: User, **kwargs):
         if not principal.has_perm("pipelines.update_pipeline_version", self):
             raise PermissionDenied
 
-        for key in ["name", "description", "external_link"]:
+        for key in ["name", "description", "external_link", "config"]:
             if key in kwargs and kwargs[key] is not None:
+                if key == "config" and self.is_schedulable:
```

Comment:
You can move that piece of code outside the for loop and treat the config directly.

Reply:
`is_schedulable` does not mean that the pipeline has a `schedule` ⚠️ It's better to check `pipeline.schedule`.